### PR TITLE
test: add e2e tests for duplicate space slugs

### PR DESCRIPTION
**Pull request by notgiorgi**
Closes: #ISSUE_NUMBER

### Description:
Added end-to-end tests to verify that duplicate space slugs are not created within the same project. The tests ensure that when spaces with identical names are created, they receive unique slugs, both at the top level and when nested under a parent space.

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Comment by notgiorgi** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/lightdash/lightdash/14622?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#14622** <a href="https://app.graphite.dev/github/pr/lightdash/lightdash/14622?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/lightdash/lightdash/14622?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#14621** <a href="https://app.graphite.dev/github/pr/lightdash/lightdash/14621?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/e2e/cypress/e2e/api/nestedSpaces.cy.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import { SEED_PROJECT } from '@lightdash/common';
+
+const apiUrl = '/api/v1';
+
+describe('Lightdash API tests for my own private spaces as admin', () => {
+    beforeEach(() => {
+        cy.login();
+    });
+    it('Should create not create duplicate slugs in the same project', () => {
```
Comment (IrakliJani) [inline packages/e2e/cypress/e2e/api/nestedSpaces.cy.ts:9]: typoo

**Review by IrakliJani — APPROVED** [review]
